File: modules/vending_model/src/main/java/com/machine/vending/model/common/CoinType.java
package com.machine.vending.model.common;

public enum CoinType {
	ONE_POUND("POUND",100),
	FIFTY_PENCE("FIFTY", 50),
	TWENTY_PENCE("TWENTY",20),
	TEN_PENCE("TEN",10),
	FIVE_PENCE("FIVE",5),
	TWO_PENCE("TWO",2),
	ONE_PENCE("ONE",1),
	UNDEFINED("UNDEFINED",0);
	
	private final String name;
	private final int value;
	private CoinType(String name, int value) {
		this.name = name;
		this.value = value;
	}
	
	public String getName() {
		return this.name;
	}
	
	public int getValue() {
		return this.value;
	}
	
	public static CoinType coinTypeByName(String name) {
		for(CoinType type: values()) {
			if(type.getName().equals(name)) {
				return type;
			}
		}
		return CoinType.UNDEFINED;
	}
	
	public static CoinType coinTypeByValue(int value) {
		for(CoinType type: values()) {
			if(type.getValue() == value) {
				return type;
			}
		}
		return CoinType.UNDEFINED;
	}
}

File: modules/vending_ui/webclient/src/model/coin_type.ts

export var useCoinType:string = 'UK'

export interface Coin {
    name: string,
    display: string,
    value: number,
    min: number, //use to set the slider
    max: number // use to set the slider
}
export interface CoinType {
    [name: string]: {[key:string]:Coin}
};

export  var coins:CoinType = {
        UK:{
            ONE_POUND:{name:"ONE_POUND", display:"1£",min:0, max:100, value:100},
            FIFTY_PENCE:{name:"FIFTY_PENCE", display:"50p",min:0,max:100, value: 50},
            TWENTY_PENCE:{name:"TWENTY_PENCE", display:"20p",min:0, max:500, value: 20},
            TEN_PENCE:{name: "TEN_PENCE", display:"10p",min:0,max:1000, value: 10},
            FIVE_PENCE:{name: "FIVE_PENCE", display:"5p",min:0, max:2000,value: 5},
            TWO_PENCE:{name:"TWO_PENCE", display:"2p",min:0, max:2000,value: 2},
            ONE_PENCE:{name:"ONE_PENCE", display:'1p',min:0, max:2000, value: 1} 
        },
        US: {}
}

export var getCoinByCountry= (country:string):{[key:string]:Coin} => {
    return coins[country];
};